docs/redesign-specs/DB_Schema/database.md

```diff
@@ -0,0 +1,199 @@
+* status : Current status of user (e.g. active, inactive, disabled, etc.)
+* autheticationId : Id of the authentication service using which a user will be verified in a multi-tenant setup. A user will specify a token (obtained from the authentication service) while invoking the REST API.
+
+****
+
+## AccessControlList
+
+1. List of users who can be given access to a particular resource (currently only for images). 
+2. An access control list (ACL) can be created by any user (a.k.a. the user creating the access control list will be its owner). 
+3. Owner of an access control list can add/remove user from the access control list. 
+4. Owner can transfer the ownership of the access control list to another user. 
+5. Owner can share an access control list with another user. 
+6. Owner can attach/detach an access control list (owned/shared by/to him) to a resource.
+7. Shared access control lists can be modified by their owner w/o approval from the user with whom the access control list was shared with.
+
+* id : Id of the access control list (primary key)
+* name : Name of the access control list 
+* ownerUserId : Id of the user who owns the ACL
+
+**NOTE:** `(name, ownerUserId) pair is unique key.`
+
+****
+
+## Image/Snapshot
+
+1. A virtual disk (a.k.a. golden image) whose clone (linked clone) is used to provision a node. 
+2. Users can upload images. 
+3. The user uploading the image will be the owner of the uploaded image.
+4. Snapshot is a special type of image that is created by deep copying the existing state of a provisioned node's image. 
+5. Image/Snapshot owner can be different from the owner of an ACL. 
+6. Ownership of the image/snapshot can be transfered to another user.
+7. Image/Snapshot can be shared to a set of users.
+8. Image/Snapshot can be made publically available to all exisiting users.
+
+* id : Id of the image (primary key)
+* name : Name of the image
+* ownerUserId : Id of the user who owns the image.
+* type : Type of image (e.g. RAW, QCOW2, etc.)
+* visibility : Access level of image (e.g. private, public, shared, etc.)
+* isSnapshot : This flag is `true` in case of snapshot and not an uploaded image
+* dataStoreId : Id of the DataStore that gives information about where this image is stored. 
+
+**NOTE:** `(name, ownerUserId) pair is unique key.`
+
+***
+
+## ProvisionedNode
+
+1. Node running an operating system from the associated clone of a parent image (aka golden image). 
+2. A user can provision multiple nodes.
+3. A clone (shallow or deep copy depending on the underlying data store's capabilities) of the parent image is created and used to provision a node.
+4. Nodes are provisioned using a provisioning engine over a network owned by the user.
+5. BMI creates an interface (physical or virtual) on the provisioning engine for the users' provisioning network.
+
+* id : Id of the provisioned node (primary key)
+* macAddress : Mac address of the provisioned node (unique key)

```

Comment:
I would rather use terms like networkAddress, and say, e.g., macAddress, some day we will move away from ethernet